src/socket.c

```diff
@@ -195,6 +195,14 @@ static int raw_connect_internal(struct trilogy_sock *sock, const struct addrinfo
         return TRILOGY_SYSERR;
     }
 
+#ifdef TCP_NODELAY
+    if (sock->base.opts.keepalive_enabled) {
```

Comment:
Should this be the nodelay option?
```suggestion
    if (sock->base.opts. tcp_nodelay_enabled) {
```